tests/test_pytypes.py

```diff
@@ -21,6 +21,45 @@ def test_iterable(doc):
     assert doc(m.get_iterable) == "get_iterable() -> Iterable"
 
 
+def test_iterable_t(doc):
+    assert doc(m.get_iterable_t) == "get_iterable_t() -> Iterable[str]"
+
+
+def test_iterable_t_overloads():
+    # Empty: First one wins.
+    list_empty = []
+    set_empty = set()
+    assert m.accept_iterable_t(list_empty) == "str"
+    assert m.accept_iterable_t(set_empty) == "str"
+    # Negative: Exhaustible iterables (e.g. iterators, generators).
+    gen_empty = (x for x in set_empty)
+    with pytest.raises(RuntimeError):
+        m.accept_iterable_t(gen_empty)
+    iter_empty = iter(list_empty)
+    with pytest.raises(RuntimeError):
+        m.accept_iterable_t(iter_empty)
+
+    # Str.
+    list_of_str = ["hey", "you"]
```

Comment:
I think it'll work with Python 2.7 if you make this `u"hey"` etc.

Reply:
I think not applicable anymore - no more Python 2 :partying_face: